test/compress/collapse_vars.js

```diff
@@ -1209,3 +1209,24 @@ collapse_vars_regexp: {
         })();
     }
 }
+
+keep_collapse_const_in_own_block_scope: {
+    options = {
+        join_vars: true,
+        loops: true
+    }
+    input: {
+        var i=2;
+        const c=5;
+        while(i--)
+            console.log(i);
+        console.log(c);
+    }
+    expect: {
+        var i=2;
+        const c=5;
+        for(;i--;)
+            console.log(i);
+        console.log(c);
+    }
+}
```

Comment:
Could you add another test where the order of the `var i=2;` and `const c=5;` lines are swapped? This will show that the for loop `var` init optimization still works.